drivers/SmartThings/jbl/src/discovery.lua

```diff
@@ -0,0 +1,104 @@
+local log = require "log"
+local discovery = {}
+
+local fields = require "fields"
+local discovery_mdns = require "discovery_mdns"
+
+local socket = require "cosock.socket"
+
+-- mapping from device DNI to info needed at discovery/init time
+local device_discovery_cache = {}
+
+local function set_device_field(driver, device)
+
+  log.info("set_device_field : dni = " .. tostring(device.device_network_id))

```

Comment:
Two things here:

1. The DNI is already a string,
2. `string.format` is a bit more idiomatic here, and takes care of handling the `tostring` in the event that the value isn't a string.

Reply:
change to

> log.info(string.format("set_device_field : %s", device.device_network_id))